vllm/v1/worker/gpu_model_runner.py

```diff
@@ -860,6 +861,53 @@ def _gather_encoder_outputs(
     def get_model(self) -> nn.Module:
         return self.model
 
+    def apply_grammar_bitmask(
+        self,
+        scheduler_output: "SchedulerOutput",
+        logits: torch.Tensor,
+    ):
+        # Serialization of np.ndarray is much more efficient than a tensor,
+        # so we receive it in that format.
+        grammar_bitmask = scheduler_output.grammar_bitmask
+        if grammar_bitmask is None:
+            return
+
+        # We receive the structured output bitmask from the scheduler, but the
+        # indices of the requests in the batch may not match the indices of
+        # the bitmask since the scheduler doesn't know how the gpu runner is
+        # ordering the requests in the batch. We need to sort the bitmask to
+        # match the order of the requests used here.
```

Comment:
Not for this PR but I wonder if we could avoid sorting every time somehow.

Reply:
yeah ... 

The runner could reply with its batch ordering, so things could be aligned as long as the same batch keeps running?